### PR TITLE
docs: add docs for group accounts

### PR DESCRIPTION
**Pull request by furi0us333**
Updating user documentation to capture additional options and commands for group accounts

### CODE REVIEW

File: docs/phylum_analyze.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ phylum analyze [options] <lockfile>
 `--filter <filter>`
 &emsp; Provide a filter used to limit the issues displayed
 
+`-g`, `--group <group_name>`
+&emsp; Group to use for analysis
```
Comment (matt-phylum) [inline docs/phylum_analyze.md:21]: All the other options for this command are documented as actions, but this one is a thing.

File: docs/phylum_group.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+---
+
+Interact with user groups
+
+```sh
+phylum group [OPTIONS] [SUBCOMMAND]
+```
+
+### Options
+`-j`, `--json`
```
Comment (matt-phylum) [inline docs/phylum_group.md:14]: This isn't clear to me. Is `--json` a parameter to `phylum group`, or is it a parameter to `phylum group list`, which is the default subcommand of `phylum group`? If I run `phylum group -j create sample`, will that produce output in JSON?

File: docs/phylum_group.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+&emsp; Produce output in json format (default: false)
+
+### Commands
+* [phylum group create](https://docs.phylum.io/docs/phylum_group_create)
+* [phylum group list](https://docs.phylum.io/docs/phylum_group_list)
```
Comment (matt-phylum) [inline docs/phylum_group.md:19]: Do we have a convention for indicating that this is the default subcommand?

File: docs/phylum_project_set-thresholds.md [hunk 4]
```diff
@@ -5,12 +5,19 @@ hidden: false
 ---
 Interactively set risk domain thresholds for a project
 ```sh
-phylum project set-thresholds <name>
+phylum project set-thresholds <name> [OPTIONS]
```
Comment (cd-work) [inline docs/phylum_project_set-thresholds.md:8]: And another instance where I'd move the options to the start. This also applies to the examples and is in line with how clap generates the help and how other unix tools behave.

File: docs/phylum_group_list.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+---
+title: phylum group list
+category: 6255e67693d5200013b1fa3e
+hidden: false
+---
+List all groups the user is part of
```
Comment (matt-phylum) [inline docs/phylum_group_list.md:6]: Nit: should it be "groups that the user is a member of" or something like that? I don't know if we normally say users are part of groups or members of groups.

File: docs/phylum_group.md (hunk 3 above)
Comment (furi0us333) [reply docs/phylum_group.md:19]: We don't currently, I have tried to just capture that via an example for those top level commands.

File: docs/phylum_group_list.md (hunk 5 above)
Comment (furi0us333) [reply docs/phylum_group_list.md:6]: I just copied what the CLI help had, so if we want to change it here we should change it there as well

File: docs/phylum_analyze.md (hunk 1 above)
Comment (furi0us333) [reply docs/phylum_analyze.md:21]: I just copied what the CLI help had, so if we want to change it here we should change it there as well

File: docs/phylum_group.md (hunk 2 above)
Comment (furi0us333) [reply docs/phylum_group.md:14]: `--json` is a parameter to `phylum group` command (see line 170)

image

File: docs/phylum_group_list.md (hunk 5 above)
Comment (cd-work) [reply docs/phylum_group_list.md:6]: Yeah I'd definitely advocate to keep the docs in line with the CLI.

File: docs/phylum_project_create.md [hunk 6]
```diff
@@ -5,12 +5,19 @@ hidden: false
 ---
 Create a new project
 ```sh
-phylum project create <name>
+phylum project create <name> [OPTIONS]
```
[the remark this answers is not in the document]
Comment (furi0us333) [reply docs/phylum_project_create.md]: fixed

File: docs/phylum_project_link.md [hunk 7]
```diff
@@ -5,12 +5,19 @@ hidden: false
 ---
 Link a repository to a project
 ```sh
-phylum project link <name>
+phylum project link <name> [OPTIONS]
```
[the remark this answers is not in the document]
Comment (furi0us333) [reply docs/phylum_project_link.md]: fixed

File: docs/phylum_project_set-thresholds.md (hunk 4 above)
Comment (furi0us333) [reply docs/phylum_project_set-thresholds.md]: fixed

File: docs/phylum_group_list.md (hunk 5 above)
Comment (furi0us333) [reply docs/phylum_group_list.md]: update in docs and CLI

**Pull request merged**